tests/Core/Tokenizers/Tokenizer/RecurseScopeMapSwitchTokenScopeTest.inc

```diff
@@ -32,3 +32,26 @@ switch((function () {
         return 'test';
 /* testSwitchClosureWithinConditionScopeCloser */
 }
+
+/* testSwitchClosureWithReturnTypeWithinCondition */
+switch((function (): string {
+    return 'bar';
+})()) /* testSwitchClosureWithReturnTypeWithinConditionScopeOpener */ {
+    case 1:
+        return 'test';
+/* testSwitchClosureWithReturnTypeWithinConditionScopeCloser */
+}
+
+/* testSwitchArrowFunctionWithinCondition */
+switch((fn() => 'bar')()) /* testSwitchArrowFunctionWithinConditionScopeOpener */ {
```

Comment:
Let's give this test and the next something to get confused over...

If using the alternative switch syntax, maybe a ternary in the return of the arrow function.
If using the "normal" switch syntax, maybe array access ?